### DISCUSSION TITLE
Accumulate logged work per Epic

### DISCUSSION THREAD

**Discussion by Ernte1893** [shridhar-tl/jira-assistant#252] [Q&A]
Hi all, 
for our projects it would be really helpfull to collect the logged time within the time frame per epic. Is there a possibility with the Jira Assistant?

**Comment by shridhar-tl** [comment]
Hi @Ernte1893, I am not very clear with your request. But as far I understand, yes, it is possible. Go to configuration of new worklog report and look at the Datasource, Log filter and JQL tab. This would let you group or filter the worklogs based on Epic and based on created date of the worklog. And as you must be already knowing, you have option to filter the worklog based on log date also.

Notice the difference here, worklog date and worklog created date filters are two different filters. Use these filters to pull the worklog accordingly.

JQL tab can be used to filter worklogs only under specific epic, project or anything else.

![image](https://user-images.githubusercontent.com/37339683/210472719-15ba8717-fb51-4cf5-bac0-df07763ed0ce.png)

If my response does not make sense or does not solve your problem, please provide more details on what exactly you are expecting for.

**Reply by Ernte1893** [reply]
Hi @shridhar-tl , 

thanks for the reply. 
Although it is still quite buggy, it is helpful. 
Question: Why is that only possible for all users who logged work? I mean, you could still specify the users in the JQL filter, so in theory it is possible?

**Reply by shridhar-tl** [reply]
I think you misunderstood. Forget about User List in this screen. Just look at the option I have highlighted. And yes, you can either add users from query, or use the inbuilt user picker within JA or show all the users who logged work within the selected timeframe.

I am not sure what you call as buggy. If you encounter any issues, please report that issue so that it can be fixed.

There are few settings which you may misunderstand to something else. So I would suggest you to watch the following video which may clear few queries: https://www.youtube.com/watch?v=WQOHrPBZZiI&list=PLxDBQLgpBhsDGppYrDqARYULukk7XFJPV&index=5

Additionally if you are new to Jira Assistant itself, You can watch following video as well: https://www.youtube.com/watch?v=TxNH1HQtiX0&list=PLxDBQLgpBhsDGppYrDqARYULukk7XFJPV&index=6

There are more videos on the same playlist which you can watch to know about individual module. Alternatively, from each module of Jira Assistant, if you click on YouTube icon from top right corner of the page, you can watch video of that specific module.

**Reply by shridhar-tl** [reply]
Also, good to mention, what is the version of Jira Assistant are you using? v2.44 is the latest version and you can update your extension if you are using old version.

I suggest it because there are huge list of bugfixes in this release which includes some related to worklog report as well.

**Comment by Ernte1893** [comment]
@shridhar-tl : Thanks for the update. 
I am using v2.44 (or to double check even the web version, which should be even newer). 

What I mean with issues: 

- You can only select the "group by epic" option if BOTH "Show all users who logged work" AND "Based on selected date range" is selected. That is a issue for me (or at least not self explainable :) )
- There is a comment when grouping by epics that says "Jira Epic Field name must be configured" : Must be configured by what? If it has to be configured to "Epic Link" then I suggest to mention that, mark it as a warning if not or rather directly change the configuration to it!

Anyway, thanks for your support.

**Reply by shridhar-tl** [reply]
Yes, your concerns are valid and there could have been better UX design. But I am a core developer and not an expert in UX. Honestly speaking,  I am good enough to understand this UI is not up to the expectation, but not proficient enough to think of what would make it better. So I have always welcome suggestions from users to make things better. 

However I would like to let you know the reason why it was done so:

**For Point # 1:**
As of now group by epic or project is only available in specific case. The reason for this is as follows:

1) When you select "Show only selected users" from "User List", it is already grouped based on the User Groups created by user. So additional sub grouping is not supported as of now.
2) When you select "Based on specific sprint" already it is grouped based on sprints. Similarly additional sub grouping is not supported as of now.

That is the reason this custom grouping based on project or epic is supported with a combination of "Show all users who logged work" and "Based on selected date range". 

**[Update] :** This point is partially fixed. Now custom grouping is available irrespective of what you select in "User List". You can see it in action from Web version or once your extension gets updated to v2.47 which is already published and is being reviewed by Web store.

**For Point # 2:**
I can by default select "Epic Link", but unfortunately in Jira these fields were customizable. This implementation was done this way based on issue reports from users way back in 2017. So that users can choose the appropriate field from settings for epic.

Once again, both of these were clarified in some of the videos I suggested. But now knowing the background, if you have any suggestions on what would make it more clear now or in future, feel free to raise a ticket with your suggestion.

I cannot assure I will have it implemented immediately, but I have always implemented valid suggestions sooner or later.

**Reply by shridhar-tl** [reply]
No need to mention, anyone is always welcome to raise a PR with improvements to the tool which is useful for others as well.
